denops/ai-review/store/config.ts

```diff
@@ -29,11 +40,31 @@ export const configSlice = createSlice({
   initialState: configInitialState,
   reducers: {
     config: (state, action: PayloadAction<ConfigState>) => {
-      // TODO: validate config
-      state.config = action.payload.config
+      const input = action.payload.config
+      validate(input)
+      state.config = input
     },
   },
 })
 
-export const modelSelector = (): string => store.getState().config.config.chat_gpt.model
-export const logDirSelector = (): string => store.getState().config.config.log_dir
+const validate = (config: Config) => {
```

Comment:
Sorry the reviews have been separated by so much time.
Since `validate` is a generic name, please change it to a function name that clearly states that it is a config validate.
Also, I think it would be better to implement it as Assertion functions.
Ref: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

```
const assertIsConfig = (config: unknown): asserts config is Config => {
  // ...
}
```

Reply:
I missed that this file is config.ts.
The function name is fine for validate or assert.

Reply:
I was thinking about how to write the code a bit by fiddling with it at hand, but it seems like a lot of work to ASSETS it, so there is no problem to merge it as it is.